src/gui/fileactivitymodel.cpp

```diff
@@ -0,0 +1,104 @@
+#include <theme.h>
+#include <QLoggingCategory>
+
+#include "fileactivitymodel.h"
+#include "activityjob.h"
+
+Q_LOGGING_CATEGORY(lcFileActivityModel, "nextcloud.gui.fileactivitymodel", QtInfoMsg)
+
+namespace OCC {
+
+FileActivity::FileActivity() = default;
+
+FileActivity::FileActivity(const QString &message, const QString &timestamp, const QIcon &icon)
+    : _message(message)
+    , _timestamp(timestamp)
+    , _icon(icon)
+{
+}
+
+QString FileActivity::message() const
+{
+    return _message;
+}
+
+QString FileActivity::timestamp() const
+{
+    return _timestamp;
+}
+
+QIcon FileActivity::icon() const
+{
+    return _icon;
+}
+
+FileActivityListModel::FileActivityListModel(QObject *parent)
+    : QAbstractListModel(parent)
+{
+    _timer.setInterval(3000);
+    _timer.setSingleShot(true);
+    _timer.start();
+
+    connect(&_timer, &QTimer::timeout, this, [this] {
+        std::vector<FileActivity> fileActivities;
+        for (int i = 0; i < 100; ++i) {
+            fileActivities.emplace_back("My message", "12:15", Theme::instance()->applicationIcon());
+        }
+        setFileActivities(fileActivities);
+    });
+}
+
+QModelIndex FileActivityListModel::addFileActivity(const FileActivity &fileActivity)
+{
+    const auto rowIndex = rowCount();
+    beginInsertRows(QModelIndex(), rowIndex, rowIndex);
+    _fileActivities.push_back(fileActivity);
+    endInsertRows();
+    return index(rowIndex, 0);
+}
+
+void FileActivityListModel::setFileActivities(const std::vector<FileActivity> &fileActivities)
+{
+    beginResetModel();
+    _fileActivities = fileActivities;
+    endResetModel();
+}
+
+int FileActivityListModel::rowCount(const QModelIndex &parent) const
+{
+    Q_UNUSED(parent)
+    return _fileActivities.size();
+}
+
+QVariant FileActivityListModel::data(const QModelIndex &index, int role) const
+{
+    if (!index.isValid()) {
+        return QVariant();
+    }
+
+    if (role != Qt::DisplayRole) {
+        return QVariant();
+    }
+
+    return QVariant::fromValue(_fileActivities[index.row()]);
+}
+
+QVariant FileActivityListModel::headerData(int section, Qt::Orientation orientation, int role) const
+{
+    return QAbstractListModel::headerData(section, orientation, role);
+}
+
+FileActivityDialogModel::FileActivityDialogModel(std::unique_ptr<ActivityJob> activityJob, QObject *parent)
+    : QObject(parent)
+    , _activityJob(std::move(activityJob))
+{
+    const QString type = "files";
+    const QString id = "00000690ocln0mwcphri";
+    _activityJob->queryActivities(type, id, 2);
```

Comment:
please better use QStringLiteral directly

Reply:
Yes, but that is from an old commit. Is not there anymore :)